Assignment6.Rmd

```diff
@@ -66,20 +70,112 @@ M2$predict1 <- predict(c.tree1, M2, type = "class")
+library('randomForest')
+
+data2 <- read.csv("student.record.csv", header = TRUE)
+#Choose numerical value from data 2
+data3 <- data2[,c(3:9,13)]
+
+# Deal with NA
+data4 <- na.omit(data3)
+data5 <- subset(data4, data4$MAJOR1_DESCR != "")
+```
+
+
+# model 1 Use raw variables
+
+```{r}
+model1 <- rpart(as.factor(MAJOR1_DESCR) ~HSGPA+ LAST_ACT_ENGL_SCORE+LAST_ACT_MATH_SCORE+LAST_ACT_READ_SCORE+LAST_ACT_SCIRE_SCORE+LAST_ACT_COMP_SCORE+as.factor(SEX), method = "class", data = data5)
+
+printcp(model1)
+
+# CP nsplit rel error  xerror      xstd
+# 1 0.02168      0   1.00000 1.00000 0.0012575
+# 2 0.01000      1   0.97832 0.97832 0.0014301
+```
+
+# model 2 Use trasformed data
+
+```{r}
+library(stringr)
+#Conclude the MAJOR1_DESCR into larger group
+data5$DIP<-word(data5$MAJOR1_DESCR, -1)
+data5$DIP[which(data5$DIP=="Fresh")]<-"BS"
+data5$DIP[which(data5$DIP=="ChemE")]<-"BSE"
+data5$DIP[which(data5$DIP=="BSChem")]<-"BS"
+data5$DIP[which(data5$DIP=="Chem")]<-"BS"
+data5$DIP[which(data5$DIP=="CompS")]<-"BSE"
+data5$DIP[which(data5$DIP=="ElecE")]<-"BSE"
+data5$DIP[which(data5$DIP=="CS")]<-"BSE"
+data5$DIP[which(data5$DIP=="Aero")]<-"BSE"
+data5$DIP[which(data5$DIP=="Sc")]<-"BSE"
+data5$DIP[which(data5$DIP=="E")]<-"BSE"
+
+#Select the top 5 category for prediction
+s1 <- sort(table(data5$DIP),decreasing = TRUE)[1:5]
+data6 <- data5[which(data5$DIP %in% rownames(s1)),]
+
+#sacle the dataset
+library(dplyr)
+library(rpart.plot)
+data6<-filter(data6,HSGPA != 0) 
+ind <- sapply(data6, is.numeric)
+data7<- as.data.frame(lapply(data6[ind], scale))
+data7 <- cbind(data7, data6[,8:9])
+
+#model2
+model2 <- rpart(as.factor(DIP) ~HSGPA+ LAST_ACT_ENGL_SCORE+LAST_ACT_MATH_SCORE+LAST_ACT_READ_SCORE+LAST_ACT_SCIRE_SCORE+LAST_ACT_COMP_SCORE+as.factor(SEX), method = "class", data = data7)
+
+printcp(model2)
+#xerror:0.89035
+
+rpart.plot(model2)
+```
+
+# model 3 Use random forest
+
 ```{r}
 
+model3 <- randomForest(as.factor(DIP) ~HSGPA+ LAST_ACT_ENGL_SCORE+LAST_ACT_MATH_SCORE+LAST_ACT_READ_SCORE+LAST_ACT_SCIRE_SCORE+LAST_ACT_COMP_SCORE+SEX, data = data7)
+
+#number of trees
+plot(model3)
+
+print(model3)
+#OOB estimate of  error rate: 52.67%
+
 ```
 
+# Conclusion
+
+According to 3 models above,
+
+Model1 : used raw data as input and got nearly 97.832% of error rate.
+
+Model2 : After combined different types of major into larger group(eg:BA,BE,BSE), and scaled the ACT score, the error rate decrese to 90%, which is much lower than the model1.
+
+This model is used to predict the diploma which student choosed based on the ACT score. The result seems reasonable. Student with lower ACT math score will choose Bachelor of Arts. Students with both high math score and high highschool GPA will tend to choose Bachelor of Sciences. Besides, gender also plays an important role in major, male will prefer BSE (Engineering), and female with high math score will choose BA and BS.
+
+Model3 : After using random forest to further improve the model, the error rate droped to 52.67%.
```

Comment:
Very nice work! I appreciate that you manipulate/change the form of data (outcome) and use different model to improve the precision of predictions and interpret some interesting findings from the models. Random forest is an effective way although out of the scale of this class, but it is great that you explore other methods for the assignment.